inngest-core/src/main/kotlin/com/inngest/signingkey/BearerToken.kt

```diff
@@ -0,0 +1,41 @@
+package com.inngest.signingkey
+
+import com.inngest.RequestHeaders
+import java.lang.NumberFormatException
+import java.security.MessageDigest
+
+val SIGNING_KEY_REGEX = Regex("""(?<prefix>^signkey-[\w]+-)(?<key>.*)""")
+
+/**
+ * Takes a signing key in the form "signkey-<env>-<hex-encoded key>" and returns "signkey-<env>-<hex-encoded sha256 of key>"
+ * Inspired by https://www.baeldung.com/sha-256-hashing-java
+ *
+ * @param signingKey signing key in the form "signkey-<env>-<hex-encoded key>"
+ * @return the hashed signing key in the form "signkey-<env>-<hex-encoded sha256 of key>"
+ * @throws InvalidSigningKeyException If signingKey is not in the form "signkey-<env>-<key>"
+ */
+@OptIn(ExperimentalStdlibApi::class)
```

Comment:
need this for both `hexToByteArray` and `toHexString`, otherwise we need to pull in other libraries or loop over and convert manually